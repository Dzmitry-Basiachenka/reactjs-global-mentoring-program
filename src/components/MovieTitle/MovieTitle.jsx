import React from 'react';
import { Container, Row, Col } from 'react-bootstrap';

import { getGenres, getYear } from '../../utils/utils.js';
import styles from './styles.module.css';

const MovieTitle = ({ movie }) => {
    return (
        <Container>
            <Row>
                <Col md={3}>
                    <img data-testid='movie-poster' className="img-fluid" src={movie.poster_path} alt={movie.title} />
                </Col>
            </Row>
            <Row>
                <Col md={3}>
                    <p className={styles.flexbox}>
                        <span data-testid='movie-title' className={styles.title}>{movie.title}</span>
                        {movie.release_date && <span data-testid='movie-year' className={styles.year}>{getYear(movie.release_date)}</span>}
                    </p>
                    {movie.genres && <p data-testid='movie-genres'>{getGenres(movie.genres)}</p>}
                </Col>
            </Row>
        </Container>
    )
}

export default MovieTitle;
