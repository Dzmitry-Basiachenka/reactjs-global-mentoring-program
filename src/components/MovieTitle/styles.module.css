.flexbox {
    display: flex;
    align-items: center;
}

.title {
    font-size: 16px;
    font-weight: 500;
    margin: 5px 90px 0px 0px;
}

.year {
    border: 1px solid #555;
    border-radius: 5px;
    margin: 5px 0px 0px 0px;
}
