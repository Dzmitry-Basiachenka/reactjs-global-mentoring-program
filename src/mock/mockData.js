export const MOCK_MOVIES = [
  {
    "id": 680,
    "title": "Pulp Fiction",
    "tagline": "Just because you are a character doesn't mean you have character.",
    "vote_average": 8.3,
    "vote_count": 10910,
    "release_date": "1994-09-10",
    "poster_path": "https://cdn.europosters.eu/image/1300/posters/pulp-fiction-cover-i1288.jpg",
    "overview": "A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.",
    "budget": 8000000,
    "revenue": 213928762,
    "genres": [
      "Thriller",
      "Crime"
    ],
    "runtime": 154
  },
];

export const MOCK_SORT_OPTIONS = [
  {
    name: 'Release date',
    value: 'release_date'
  },
  {
    name: 'Title',
    value: 'title'
  },
];

export const MOCK_GENRES = [
  'Crime',
  'Documentary',
  'Horror',
  'Comedy'
];
