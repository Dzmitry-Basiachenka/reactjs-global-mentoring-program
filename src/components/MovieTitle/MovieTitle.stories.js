import MovieTitle from './MovieTitle.jsx';
import { MOCK_MOVIES } from '../../mock/mockData.js';

export default {
  component: MovieTitle
};

export const Default = {
  args: {
    movie: MOCK_MOVIES[0],
  },
  parameters: {
  }
};
