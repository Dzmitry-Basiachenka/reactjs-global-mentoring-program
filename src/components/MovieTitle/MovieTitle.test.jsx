import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render, screen } from '@testing-library/react'

import MovieTitle from './MovieTitle.jsx';
import { MOCK_MOVIES } from '../../mock/mockData.js';

describe('MovieTitle', () => {

  afterEach(cleanup);

  it('should match snapshot', () => {
    const movieTitle = renderer
      .create(
        <MovieTitle
          movie={MOCK_MOVIES[0]}
        />
      )
      .toJSON();

    expect(movieTitle).toMatchSnapshot();
  });

  it('render correctly', () => {
    const movie = MOCK_MOVIES[0];

    render(
      <MovieTitle
        movie={movie}
      />
    );

    const image = screen.getByTestId('movie-poster');
    expect(image.getAttribute('src')).toEqual(movie.poster_path);
    expect(image.getAttribute('alt')).toEqual(movie.title);

    expect(screen.getByTestId('movie-title')).toHaveTextContent(movie.title);
    expect(screen.getByTestId('movie-year')).toHaveTextContent('1994');
    expect(screen.getByTestId('movie-genres')).toHaveTextContent('Thriller, Crime');
  });

});
