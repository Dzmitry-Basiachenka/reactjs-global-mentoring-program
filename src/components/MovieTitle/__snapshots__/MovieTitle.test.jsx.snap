// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTitle should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-3"
    >
      <img
        alt="Pulp Fiction"
        className="img-fluid"
        data-testid="movie-poster"
        src="https://cdn.europosters.eu/image/1300/posters/pulp-fiction-cover-i1288.jpg"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-3"
    >
      <p
        className="flexbox"
      >
        <span
          className="title"
          data-testid="movie-title"
        >
          Pulp Fiction
        </span>
        <span
          className="year"
          data-testid="movie-year"
        >
          1994
        </span>
      </p>
      <p
        data-testid="movie-genres"
      >
        Thriller, Crime
      </p>
    </div>
  </div>
</div>
`;
